client/components/banner-notice/index.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { Flex, FlexItem, Icon, Notice, Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies.
+ */
+import './styles.scss';
+import classNames from 'classnames';
+
+/**
+ * Props for the BannerNotice component.
+ *
+ * @typedef {Object} BannerNoticeProps
+ * @property {Icon.IconType<unknown>} icon The icon to display.
+ */
+interface BannerNoticeProps extends Notice.Props {
+	icon?: Icon.IconType< unknown >;
```

Comment:
@Jinksi does this look right to you? 

My `BannerNoticeProps` extends the default `Notice.Props` and adds a new optional `icon` prop that is supposed to be a `<Icon>` supported type. 


`IconType` is defined like this:  

<img width="817" alt="Screenshot 2023-03-21 at 2 42 36 pm" src="https://user-images.githubusercontent.com/8490476/226518716-049d8a38-c8c2-4403-bc1f-3e4934f274a3.png">

The `unknown` bit indicates that the type is not important (at least that's how its been explained to me)? Is that right? Should this be something else? 

I tried using `P` instead of unknown too but it wasn't happy with that. 

<img width="703" alt="Screenshot 2023-03-21 at 2 54 39 pm" src="https://user-images.githubusercontent.com/8490476/226522071-8b274b02-2a50-496e-afc7-379af64688fe.png">

Reply:
This is getting into some parts of TS that I'm not too familiar with. But, this looks suitable to me.

`Icon.IconType< unknown >;` I think is saying "We don't know what is being passed as `P` (the props of the component being passed to `icon={}`), so make sure you check them."

Whereas `Icon.IconType< any >;` would tell TS not to bother checking the props of the `icon={}` component.

Testing the type safety by passing incorrect types results in the correct behaviour. 👍

In the below example, `in` === ❌, `info` === ✅ (`info` is from the type `Dashicon.Icon`).

<img width="354" alt="image" src="https://user-images.githubusercontent.com/3147296/226755059-157aba1c-32ca-42bb-8bc9-c84806463ef1.png">